oe_webtools_analytics/oe_webtools_analytics.behat.inc

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * @file
+ * Contains \OEParagraphsSubContext.
+ */
+
+declare(strict_types = 1);
+
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use Drupal\DrupalExtension\Context\DrupalSubContextBase;
+use Drupal\oe_webtools_analytics\Utils\WebtoolsAnalyticsIndex;
+
+/**
+ * Behat step definitions for testing the OE Webtools Analytics module.
+ */
+class OEWebtoolsAnalyticsSubContext extends DrupalSubContextBase {
+
+  /**
+   * Setup demo site.
+   *
+   * @BeforeScenario @webtoolsAnalytics
+   */
+  public function setupWebtoolsAnalytics(BeforeScenarioScope $scope) {
+    \Drupal::service('module_installer')->install(['oe_webtools_analytics']);
+  }
+
+  /**
+   * Checking if the settings on the page match the value on config.
+   *
+   * @Then the rendered script should contains the :arg1 with the same value as the configuration set in settings file
+   */
+  public function theSiteIdRenderedOnThePageShouldBeTheSameAsInSettingsFile($siteIdLabel) {
+    // Getting the current settings.
+    $settingsList = $this->getAnalyticsSettingsFromPage();
+
+    // The siteId from the settings.php file.
+    $settingsSiteId = \Drupal::configFactory()->get('oe_webtools.analytics')->get(WebtoolsAnalyticsIndex::SITE_ID);
+
+    // The rendered siteId.
+    $onPageSiteId = $settingsList[$siteIdLabel];
+
+    // Message to be displayed in case of failure.
+    $failMsg = "The siteID on settings is $settingsSiteId which differ to the rendered siteID = $onPageSiteId";
+
+    PHPUnit_Framework_Assert::assertEquals($onPageSiteId, $settingsSiteId, $failMsg);
+  }
+
+  /**
+   * Set a new config value for siteID.
+   *
+   * @Then I set the SiteId with :arg1 in settings file
+   */
+  public function setSideIdInSettingsFile($siteIdValue) {
+    \Drupal::configFactory()->getEditable('oe_webtools.analytics')
```

Comment:
This is storing the config, but not setting it in the settingsfile. Better update the function name to reflect the implementation.